Source/Config/Main.lua

```diff
@@ -0,0 +1,50 @@
+Auctionator.Config.Option = {
+  DEBUG = "debug",
+  MAILBOX_TOOLTIPS = "mailbox_tooltips",
+  VENDOR_TOOLTIPS = "vendor_tooltips",
+  AUCTION_TOOLTIPS = "auction_tooltips",
+  ENCHANT_TOOLTIPS = "enchant_tooltips",
+  SHOW_LISTS = "show_lists",
+}
+
+local defaults = {
+  [Auctionator.Config.Option.DEBUG] = false,
+  [Auctionator.Config.Option.MAILBOX_TOOLTIPS] = true,
+  [Auctionator.Config.Option.VENDOR_TOOLTIPS] = true,
+  [Auctionator.Config.Option.AUCTION_TOOLTIPS] = true,
+  [Auctionator.Config.Option.ENCHANT_TOOLTIPS] = true,
+  [Auctionator.Config.Option.SHOW_LISTS] = true,
+}
+
+function validOption(name)
+  for _, option in pairs(Auctionator.Config.Option) do
+    if option == name then
+      return true
+    end
+  end
+  return false
+end
+
+function Auctionator.Config.Set(name, value)
+  if AUCTIONATOR_CONFIG == nil then
+    error("AUCTIONATOR_CONFIG not initialized")
+  elseif not validOption(name) then
+    error("Invalid option '" .. name .. "'")
+  else
+    AUCTIONATOR_CONFIG[name] = value
+  end
+end
+
+function Auctionator.Config.Reset()
+  AUCTIONATOR_CONFIG = {}
+end
+
+function Auctionator.Config.Get(name)
+  if AUCTIONATOR_CONFIG == nil then
+    return nil
+  elseif AUCTIONATOR_CONFIG[name] == nil then
```

Comment:
Do we want to initialize to the default values in an OnLoad function if the entry doesn't exist? Probably not a huge deal but saves us some complexity here to be able to assume a value is always set. (For that matter, we may want to do the same thing in Reset?)